### PR TITLE
Add users details page (only admins can see)

### PR DESCRIPTION
**Pull request by amnindersingh12**
- Admin user can view all users and their role.
- Only admins can view all users and their role.
- Refactor Views, user can see his own `role` alongside his `email`
- Fix Callbacks issue
- (use `after_create` instead of `after_save`, because after_save is invoked when an object is created and updated. `after_commit` is called to create, update and destroy. `after_create` is only called when creating an object )

### CODE REVIEW

File: app/helpers/courses_helper.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ def current_subscription(id)
     current_user.subscriptions.find_by(course_id: id)
   end
 
-  def have_superusers(id)
+  def superuser?(id)
     Course.all.pluck(:superuser_id).uniq.include?(id)
   end
```
Comment (swatijadhav) [inline app/helpers/courses_helper.rb]: This should be present in the model

File: app/models/ability.rb [hunk 2]
```diff
@@ -10,11 +10,17 @@ def initialize(user)
...
 
-    can(%i[enroll read read mark_as], :all)
+    can(%i[enroll read mark_as], :all) if user.member?
 
     can(:manage, Course, user_id: user.id) if user.admin?
 
-    can(%i[update], Course, superuser_id: user.id)
+    can(:update, Course, superuser_id: user.id)
+
+    can(%i[read mark_as enroll], Course)
```
Comment (swatijadhav) [inline app/models/ability.rb]: @amnindersingh12 While defining abilities:
the ones with less restrictions should come at top and the one more with more restrictions should go at end

For example:
```
    can :read, Post, public: true

    return unless user.present?  # additional permissions for logged in users (they can read their own posts)
    can :read, Post, user: user

    return unless user.admin?  # additional permissions for administrators
    can :read, Post # admin can read all posts by all users
```

File: app/helpers/courses_helper.rb (hunk 1 above)
Comment (amnindersingh12) [reply app/helpers/courses_helper.rb]: Okay

File: app/models/ability.rb (hunk 2 above)
Comment (amnindersingh12) [reply app/models/ability.rb]: Okay

**Pull request merged**